### PR TITLE
Feature/add product photo

### PR DESCRIPTION
**Pull request by MaxAstin**

### CODE REVIEW

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/createmenuproduct/AddMenuProductViewState.kt [hunk 1]
```diff
@@ -19,32 +20,27 @@ data class AddMenuProductViewState(
...
-    val hasError: Boolean?,
     val categoryLabel: String,
     @StringRes val categoryHint: Int?,
     val isVisibleInMenu: Boolean,
     val isVisibleInRecommendation: Boolean,
-    val categoriesErrorBorder: BorderStroke?,
-    val photoBlock: PhotoBlock,
-    val selectableCategoryList: List<CategoryItem>
+    val categoriesBorder: BorderStroke?,
+    val selectableCategoryList: List<CategoryItem>,
```
Comment (xidealo) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/createmenuproduct/AddMenuProductViewState.kt:28]: Immutable list?

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageDefaults.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+object CropImageDefaults {
+
+    fun options(): CropImageOptions {
+        return CropImageOptions(
+            imageSourceIncludeCamera = false,
+            cropShape = CropImageView.CropShape.RECTANGLE,
+            showProgressBar = false,
+            autoZoomEnabled = false,
+            aspectRatioX = 1000,
```
Comment (xidealo) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageDefaults.kt:16]: lets extract magic numbers

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageFragment.kt [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            imageContent = ImageContent(uri = state.uri?.toUri())
+        )
+    }
+
+    @Composable
+    override fun Screen(state: CropImageViewState, onAction: (CropImage.Action) -> Unit) {
+        AdminScaffold(
+            title = stringResource(R.string.title_add_menu_product),
+            backActionClick = {
+                findNavController().popBackStack()
```
Comment (xidealo) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageFragment.kt:91]: onAction

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageFragment.kt [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+                            brush = SolidColor(
+                                value = AdminTheme.colors.main.primary
+                            )
+                        ),
+                        colors = IconButtonDefaults.iconButtonColors(
+                            containerColor = AdminTheme.colors.main.secondary,
+                            contentColor = AdminTheme.colors.main.primary
+                        ),
+                        onClick = {
+                            cropImageView?.rotateImage(90)
```
Comment (xidealo) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageFragment.kt:112]: Magic number

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageFragment.kt [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+                    }
+                }
+            )
+        }
+    }
+
+    override fun handleEvent(event: CropImage.Event) {
+        when (event) {
+            CropImage.Event.CropImage -> {
+                cropImageView?.croppedImageAsync(saveCompressQuality = 100)
```
Comment (xidealo) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageFragment.kt:166]: magic number

File: domain/src/main/java/com/bunbeauty/domain/feature/menu/addmenuproduct/CreateMenuProductUseCase.kt [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        val description: String,
+        val comboDescription: String,
+        val imageUri: String?,
+        val isVisible: Boolean,
+        val isRecommended: Boolean,
+        val categories: List<SelectableCategory>
+    )
+
+    suspend operator fun invoke(params: Params) {
+        val name = params.name.takeIf { it.isNotBlank() } ?: throw MenuProductNameException()
```
Comment (xidealo) [inline domain/src/main/java/com/bunbeauty/domain/feature/menu/addmenuproduct/CreateMenuProductUseCase.kt:41]: rename it to name

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageDefaults.kt (hunk 2 above)
Comment (MaxAstin) [reply app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageDefaults.kt:16]: Is it magic numbers? 🤔
I don't think so

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/createmenuproduct/AddMenuProductViewState.kt (hunk 1 above)
Comment (MaxAstin) [reply app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/createmenuproduct/AddMenuProductViewState.kt:28]: fixed

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageFragment.kt (hunk 3 above)
Comment (MaxAstin) [reply app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageFragment.kt:91]: fixed

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageFragment.kt (hunk 4 above)
Comment (MaxAstin) [reply app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageFragment.kt:112]: fixed

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageFragment.kt (hunk 5 above)
Comment (MaxAstin) [reply app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/menulist/cropimage/CropImageFragment.kt:166]: fixed

File: data/src/main/java/com/bunbeauty/data/repository/PhotoRepository.kt [hunk 7]
```diff
@@ -31,7 +47,63 @@ class PhotoRepository @Inject constructor() : PhotoRepo {
...
         photoListCache = null
     }
+
+    private fun Uri.toBitmap(): Bitmap? {
+        return context.contentResolver.openInputStream(this).use { inputStream ->
+            BitmapFactory.decodeStream(inputStream)
+        }
+    }
+
+    private fun Uri.compressQuality(): Int {
```
[the remark this answers is not in the document]
Comment (MaxAstin) [reply data/src/main/java/com/bunbeauty/data/repository/PhotoRepository.kt:82]: fixed

File: domain/src/main/java/com/bunbeauty/domain/feature/menu/addmenuproduct/CreateMenuProductUseCase.kt (hunk 6 above)
Comment (MaxAstin) [reply domain/src/main/java/com/bunbeauty/domain/feature/menu/addmenuproduct/CreateMenuProductUseCase.kt:41]: fixed

**Pull request merged**